### PR TITLE
Rails 5.2 support

### PR DESCRIPTION
**Pull request by fiestacasey**
Handled this deprecation warning: 
> Returning `false` in Active Record and Active Model callbacks will not implicitly halt a callback chain in Rails 5.1. To explicitly halt the callback chain, please use `throw :abort` instead.

### CODE REVIEW

File: spec/lib/state_machine_spec.rb [hunk 1]
```diff
@@ -44,10 +44,10 @@
     expect(test.second_state_at).not_to be_nil
 
     expect(test.has_been_made_second_state?).to be true
-    expect{ test.has_been_made_fourth_state? }.to raise_error(NotImplementedError)
+    expect{ test.has_been_made_fifth_state? }.to raise_error(NotImplementedError)
```
Comment (euclideansphere) [inline spec/lib/state_machine_spec.rb]: why this change?

Comment (fiestacasey) [reply spec/lib/state_machine_spec.rb]: I added fourth_state_at to the model so that i could assert it wasn't getting set when the callback fails in the other test. That then caused this to not raise because having the _at column is what is required to generate the has_been_made function
